### PR TITLE
Logstash connect to external Elasticsearch

### PR DESCRIPTION
**Pull request by kaisecheng**
This PR adds doc for connecting external Elaticsearch

### CODE REVIEW

File: pkg/controller/logstash/env_test.go [hunk 1]
```diff
@@ -162,3 +163,60 @@ func Test_getEnvVars(t *testing.T) {
...
+			wantErr:  false,
+		},
+		{
+			name:     "Non-HTTPS URL",
+			url:      "http://some.gcp.cloud.es.io",
+			expected: "http://some.gcp.cloud.es.io",
+			wantErr:  false,
+		},
+		{
+			name:     "Non-HTTPS without port",
```
Comment (robbavey) [inline pkg/controller/logstash/env_test.go:193]:
```suggestion
			name:     "Non-HTTPS with port",
```

File: docs/orchestrating-elastic-stack-applications/logstash.asciidoc [hunk 2]
```diff
@@ -367,6 +367,48 @@ spec:
...
+You can reference a Secret instead of an Elasticsearch cluster in the `elasticsearchRefs` section through the `secretName` attribute:
+
+[source,yaml,subs="attributes,callouts"]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: external-es-ref
+stringData:
+  url: https://abcd-42.xyz.elastic-cloud.com <1>
```
Comment (robbavey) [inline docs/orchestrating-elastic-stack-applications/logstash.asciidoc:383]: I think we should add a note somewhere that we add the default port of `443` to `https` URLs, as this is different behavior to standard logstash plugins, and different to how non-cloud Elasticsearch works out of the box

**Review by robbavey — COMMENTED** [review]
LGTM - I have a documentation suggestion, and a nit on test naming

File: docs/orchestrating-elastic-stack-applications/logstash.asciidoc [hunk 3]
```diff
@@ -367,6 +367,48 @@ spec:
...
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+    - clusterName: prod-es
+      secretName: external-es-ref <5>
+  monitoring:
+    metrics:
+      elasticsearchRefs:
+      - secretName: external-es-ref <5>
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/logstash.asciidoc:401]: Assuming a user specifies a URL without explicit port in the secret, then the changes in this PR append 443 by default for the environment. However Beats also assume 9200 by default and while the Logstash side is fixed, the monitoring will still not work until the user fixes the `url` field in the secret to include a port.

**Review by pebrc — COMMENTED** [review]
I think I am slightly -1 on the port appending. As explained in my comment it does fix the problem only for Logstash and not for any of the other stack applications e.g. the Beats used for monitoring. I think I would favour calling out explicitly in the doc that the default port expected is 9200 and any other port has to be configured explicitly via the url.

**Comment by kaisecheng** [comment]
Kibana assumes 443. Beats and Logstah assume 9200. Not sure what the default port is for others. The stack monitoring is not consistent.
+1 for calling out explicitly in doc. Instead of explaining the assumption of default port, having a note to ask for always add the port in URL may be simpler.

**Comment by kaisecheng** [comment]
@robbavey @pebrc I have removed the port appending. Now the PR only updates the doc. It is ready for review.

File: docs/orchestrating-elastic-stack-applications/logstash.asciidoc [hunk 4]
```diff
@@ -367,6 +367,50 @@ spec:
...
+----
+
+<1> The URL to reach the Elasticsearch cluster.
+<2> The username of the user to be authenticated to the Elasticsearch cluster.
+<3> The password of the user to be authenticated to the Elasticsearch cluster.
+<4> The CA certificate in PEM format to secure communication to the Elasticsearch cluster (optional).
+<5> The `secretName` and `name` attributes are mutually exclusive, you have to choose one or the other.
+
+NOTE: Please always specify the port in URL when connect to external Elasticsearch Cluster.
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/logstash.asciidoc:413]:
```suggestion
NOTE: Please always specify the port in URL when connecting to an external Elasticsearch Cluster.
```

**Review by pebrc — APPROVED** [review]

**Review by robbavey — APPROVED** [review]
LGTM

**Comment by pebrc** [comment]
@kaisecheng should I merge this?

**Comment by kaisecheng** [comment]
@pebrc Yes please!

**Pull request merged**